force-app/main/default/classes/RecurrenceService.cls

```diff
@@ -0,0 +1,14 @@
+public with sharing class RecurrenceService {
+    @TestVisible
+    private RecurrenceFactoryInterface factory = new RecurrenceFactory();
```

Comment:
Was the idea that this could be public, so people could provide their own implementation of RecurrenceFactoryInterface?

Reply:
Yes in order to provide there own iterators they'd have to also override the factory

Reply:
Right on - should we just make it public now - or is there a reason for it to be private.

Reply:
We can make it public but since we aren't global and since we don't have a need and aren't quite sure how we will allow people to bring their own factory, maybe we just leave it private for now? Doesn't matter really as long as it isn't global though, so up to you.

Reply:
We can go private for now, as long as we're keeping that future case in mind.  Forgive my enthusiasm, I'm trying to give it out before we've even built it :)